### ISSUE TITLE
[Security] Vulnerability of Low severity in "@storybook/addon-info > marksy > marked"

### ISSUE THREAD

**Issue by Bazze** [storybookjs/storybook#7842]
**Describe the bug**
New vulnerability discovered in July in a sub dependency: `@storybook/addon-info > marksy > marked`

This is in version `5.1.11` of `@storybook/addon-info`.

https://www.npmjs.com/advisories/1076

**Screenshots**
<img width="564" alt="Screenshot 2019-08-22 at 14 29 24" src="https://user-images.githubusercontent.com/1932208/63514757-9f447a80-c4e9-11e9-85b7-98c2be2a6c34.png">

**System**
```
Environment Info:
  npmPackages:
    @storybook/addon-actions: ^5.1.11 => 5.1.11
    @storybook/addon-backgrounds: ^5.1.11 => 5.1.11
    @storybook/addon-info: ^5.1.11 => 5.1.11
    @storybook/addon-knobs: ^5.1.11 => 5.1.11
    @storybook/addon-links: ^5.1.11 => 5.1.11
    @storybook/addon-viewport: ^5.1.11 => 5.1.11
    @storybook/react: ^5.1.11 => 5.1.11
```

**Comment by shilman** [comment]
`Addon-info` is being superceded by `addon-docs`, which fixes a bunch of bugs and is easier to maintain. Please give it a try! https://medium.com/storybookjs/storybook-docspage-e185bc3622bf

**Comment by ZebraFlesh** [comment]
Pointing to a beta release is not really an appropriate resolution for a security issue. Can we just get an updated 5.1.x release with remediated dependencies?

**Comment by shilman** [comment]
@ZebraFlesh PRs welcome

**Comment by ZebraFlesh** [comment]
I dug into this a bit and found the following:
* upgrading storybook's marksy dependency on marked generates some test failures
* using the marked demo, the output seems fine ([example](https://marked.js.org/demo/?outputType=html&text=%3Cdiv%3E%0A%20%20%3Cfont%20color%3D%22red%22%3E%20this%20is%20root%20red%20text%20%3C%2Ffont%3E%0A%0A%20%20*%20list%201%0A%20%20*%20list%202%0A%20%20*%20%3Cfont%20color%3D%22blue%22%3Elist%20blue%20text%3C%2Ffont%3E%0A%20%20*%20list%203%0A%20%20%20%20*%20%3Cfont%20color%3D%22green%22%3Etwo%20depth%20green%20text%3C%2Ffont%3E%0A%3C%2Fdiv%3E&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20true%2C%0A%20%22smartypants%22%3A%20true%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=0.7.0))

This leads me to believe that the renderers that marksy supplies to marked are bad, but I lack project familiarity to determine what the problem is.

**Comment by one-thunder** [comment]
Heyo, this is still present in `@storybook/addon-info": "~5.2.1` can we get just fix related to that regexp DoS issue. I know there is a proposal to just move to `addon-docs` but at the project I am working on, there is a sceptical approach to new libraries (yeah I know how it sounds). So just rising this issue once again as I saw it already being marked as innactive

**Comment by 1Jesper1** [comment]
Fix please

**Comment by shilman** [comment]
If anybody wants to issue a PR for a fix, I'm happy to get it merged. In the meantime, `addon-docs` has been released for two months and is getting better every day. Here's a good example of a production system using it:

https://reaviz.io/?path=/story/docs-intro--page

**Comment by fralewsmi** [comment]
Related: https://github.com/storybookjs/marksy/issues/78

**Comment by shilman** [comment]
Jiminy cricket!! I just released https://github.com/storybookjs/storybook/releases/tag/v5.3.0-rc.3 containing PR #9234 that references this issue. Upgrade today to try it out!

You can find this prerelease on the `@next` NPM tag.

Closing this issue. Please re-open if you think there's still more to do.

**Comment by gohyifan** [comment]
Hi @shilman, I'm facing the same issue in `npm audit`. This is in version `^5.3.18` of `@storybook/addon-info`.

```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ marked                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.6.2                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @storybook/addon-info [dev]                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @storybook/addon-info > marksy > marked                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/812                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by shilman** [comment]
@gohyifan https://github.com/storybookjs/storybook/issues/7842#issuecomment-523951188
